torch/csrc/jit/codegen/cuda/lower_index.cpp

```diff
@@ -197,15 +197,27 @@ void IndexLowering::handle(const TernaryOp* top) {
 }
 
 void IndexLowering::handle(const IndexSelectOp* sop) {
-  const auto indices = lowerSrcIndex(sop->input(1), sop->output(0));
+  auto lowered_index = lowerSrcIndex(sop->input(1), sop->output(0));
```

Comment:
If we cast before lowering index, would it free us from adding the type promotion to IrBuilder? It looks hacky to me that we only enable type promotion for integers but not floating points.

Reply:
I meant to insert a cast op to convert `sop->input(1)` into `GpuLower::current()->kernel()->indexType()`, would this work?

Reply:
`sop->input(1)` is just a `TensorView`, so casting itself doesn't seem to make sense. We could change how we generate index expressions and make sure they are of type nvfuser_index_t, which would mean we would need a lot more significant changes.

Reply:
OK, I see. Then I am fine with the current status.